community/kernel/src/main/java/org/neo4j/kernel/recovery/CorruptedLogsTruncator.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.recovery;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.util.function.LongConsumer;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.neo4j.io.ByteUnit;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.fs.StoreChannel;
+import org.neo4j.kernel.impl.transaction.log.LogPosition;
+import org.neo4j.kernel.impl.transaction.log.PhysicalLogFiles;
+
+import static java.lang.String.format;
+
+/**
+ * Transaction log truncator used during recovery to truncate all the logs after some specified position, that
+ * recovery threats as corrupted or non-readable.
+ * Transaction log file specified by provided log position will be truncated to provided length, any
+ * subsequent files will be removed.
+ * Any removed or modified log content will be stored in separate corruption logs archive for further analysis and as
+ * an additional safety option to have the possibility to fully restore original logs in a faulty case.
+ */
+public class CorruptedLogsTruncator
+{
+    public static final String CORRUPTED_TX_LOGS_FOLDER_NAME = "corrupted-tx-logs";
```

Comment:
Should this rhyme with how the transaction log files are currently called? They're currently called `neostore.transaction.db.v???`